src/index.tsx

```diff
@@ -155,3 +155,44 @@ export const init = (opts: SdkOptions): SdkHandle => {
     },
   }
 }
+
+type CaptureSteps = 'intro' | 'front' | 'back'
+type CaptureStepActions = 'NEXT_CAPTURE_STEP' | 'RESET_CAPTURE_STEP'
+
+type MachineSpec<S extends string, A extends string> = {
+  initialState: S
+  states: Partial<Record<S, Partial<Record<A, S>>>>
+}
+
+const buildReducer = <S extends string, A extends string>(
+  spec: MachineSpec<S, A>
+) => (currentState: S, action: A): S => {
+  const stateTransitions = spec.states[currentState]
+
+  if (stateTransitions) {
+    const nextState = stateTransitions[action]
+
+    if (nextState) {
+      return nextState as S
+    }
+
+    return currentState
+  }
+
+  return currentState
+}
+
+const spec: MachineSpec<CaptureSteps, CaptureStepActions> = {
+  initialState: 'intro',
+  states: {
+    intro: {
+      NEXT_CAPTURE_STEP: 'front',
+    },
+    front: {
+      NEXT_CAPTURE_STEP: 'front',
+      RESET_CAPTURE_STEP: 'intro',
+    },
+  },
+}
+
+const reducer = buildReducer(spec)
```

Comment:
Where do you use this reducer?

Reply:
Interesting :thinking: IMO the whole diff in this file was a mistake by accident